gridstatus/caiso/caiso.py

```diff
@@ -1010,6 +1055,147 @@ def _add_forecast_publish_time(
+            try:
+                return self.get_hasp_renewable_forecast_hourly(
+                    pd.Timestamp.now(tz=self.default_timezone) + pd.Timedelta(hours=2),
+                )  # NB: This is a hack to get the latest forecast
+            except KeyError:
+                return self.get_hasp_renewable_forecast_hourly(
+                    pd.Timestamp.now(tz=self.default_timezone) + pd.Timedelta(hours=1),
+                )
+
+        df = self.get_oasis_dataset(
+            dataset="hasp_renewable_forecast_hourly",
+            date=date,
+            end=end,
+            verbose=verbose,
+            raw_data=False,
+        )
+        return self._handle_renewable_forecast(
+            df,
+            publish_time_offset=pd.Timedelta(minutes=90),
+        )
+
+    def get_rtd_renewable_forecast(
+        self,
+        date: str | pd.Timestamp,
+        end: str | pd.Timestamp | None = None,
+        verbose: bool = False,
+    ) -> pd.DataFrame:
+        """Get RTD renewable forecast from CAISO.
+
+        Args:
+            date (str | pd.Timestamp): date to return data
+            end (str | pd.Timestamp | None, optional): last date of range to return data.
+            verbose (bool, optional): print out url being fetched. Defaults to False.
+
+        Returns:
+            pandas.DataFrame: A DataFrame of RTD renewable forecast
+        """
+        if date == "latest":
+            return self.get_rtd_renewable_forecast(
+                pd.Timestamp.now(tz=self.default_timezone),
+            )
+
+        df = self.get_oasis_dataset(
+            dataset="rtd_renewable_forecast",
+            date=date,
+            end=end,
+            verbose=verbose,
+            raw_data=False,
+        )
+        return self._handle_renewable_forecast(
+            df,
+            publish_time_offset=pd.Timedelta(minutes=10),
+        )
+
+    def get_rtpd_renewable_forecast(
+        self,
+        date: str | pd.Timestamp,
+        end: str | pd.Timestamp | None = None,
+        verbose: bool = False,
+    ) -> pd.DataFrame:
+        """Get RTPD renewable forecast from CAISO.
+
+        Args:
+            date (str | pd.Timestamp): date to return data
+            end (str | pd.Timestamp | None, optional): last date of range to return data.
+            verbose (bool, optional): print out url being fetched. Defaults to False.
+
+        Returns:
+            pandas.DataFrame: A DataFrame of RTPD renewable forecast
+        """
+        if date == "latest":
+            return self.get_rtpd_renewable_forecast(
+                pd.Timestamp.now(tz=self.default_timezone),
+            )
+
+        df = self.get_oasis_dataset(
+            dataset="rtpd_renewable_forecast",
+            date=date,
+            end=end,
+            verbose=verbose,
+            raw_data=False,
+        )
+        return self._handle_renewable_forecast(
+            df,
+            publish_time_offset=pd.Timedelta(minutes=45),
```

Comment:
When possible, I think we should use the official CAISO documentation for publication times. The schedules can be found at ATLAS REFERENCE > Publications > OASIS Publications Schedule.

<img width="414" alt="Screenshot 2025-06-16 at 12 28 41" src="https://github.com/user-attachments/assets/0fb48705-5c01-41ed-a474-5a2af561d047" />

The OASIS publications schedule says this is published 22.5 minutes ahead of time. 

<img width="1386" alt="Screenshot 2025-06-16 at 12 31 30" src="https://github.com/user-attachments/assets/55c91172-6ebc-4159-972d-3b6eec241795" />